src/System.Management.Automation/engine/ManagementObjectAdapter.cs

```diff
@@ -1060,7 +1081,11 @@ internal class ManagementObjectAdapter : ManagementClassApdapter
             {
                 foreach (PropertyData property in wmiObject.Properties)
                 {
-                    members.Add(new PSProperty(property.Name, this, wmiObject, property) as T);
+                    T member = new PSProperty(property.Name, this, wmiObject, property) as T;
+                    if (predicate == null || predicate(member))
```

Comment:
Is it practical to have a "predicate" that creates the property instead - so you can avoid the allocation if the member won't be added?

Reply:
I like that idea!

Reply:
Question is if we have all the context when writing the delegate. I'll take a look.

Reply:
I change the predicates to only work on member names. That allowed me to skip most, but not all allocations. Looks quite good now in general.